### ISSUE TITLE
key/property rules and ObjectPattern

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12048]
**Tell us about your environment**

* **ESLint Version:** 6.1.0
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 6,
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**

There are 4 rules that report `ObjectPattern` properties, it's probably a bug because there are no examples in the documentation and there are no test cases.

```js
/*eslint quote-props: "error"*/
/*eslint key-spacing: "error"*/
/*eslint no-useless-computed-key: "error"*/
/*eslint computed-property-spacing: "error"*/

({
    a : foo,
    ["b" ]: bar
} = baz)
```

**What did you expect to happen?**

No warnings.

**What actually happened? Please include the actual, raw output from ESLint.**

4 warnings, one for each of the rules.

**Are you willing to submit a pull request to fix this bug?**

Yes, for whatever is decided.

`quote-props` and `key-spacing` are not working well at the moment (see PRs #12046 and #12047), I guess these fixes should be applied to skip `ObjectPattern` for now. A possible enhancements could be to add options later.

`no-useless-computed-key` and `computed-property-spacing` might work well already. Possible actions could be:

* Fix to ignore `ObjectPattern`.
* Or, consider this as a default behavior feature (it already is) - just modify the docs and add test cases.
* Maybe add options instead.

**Comment by kaicataldo** [comment]
I think we'll need to evaluate each of these rules separately. It seems like all these rules should apply to this syntax, but they should probably be individually configurable.

`key-spacing`, `no-useless-computed-key`, `computed-property-spacing`, all seem fine in their default state, I think `key-spacing` should apply here and the other two don't apply unless someone uses a computed identifier (which apparently is valid syntax - TIL!). `quote-props` does seem like an outlier - I've never actually seen that out in the wild - but if it's legal, maybe it should also be on by default (with an option to turn it off?)

Example code:
```js
const foo = { a: 'b' };

const { 'a': bar } = foo;
const { ['a']: baz } = foo;
```

**Comment by nzakas** [comment]
@mdjermanovic what do you want to do with this issue?

**Comment by mdjermanovic** [comment]
For `computed-property-spacing`, it makes the most sense to keep the current behavior and just document it (https://github.com/eslint/eslint/pull/14004 and https://github.com/eslint/eslint/pull/15423). and add tests (https://github.com/eslint/eslint/pull/15424).

**Comment by nzakas** [comment]
@mdjermanovic do you want to finish this issue up?

**Comment by mdjermanovic** [comment]
Yes, I'll check `no-useless-computed-key`, and then update the documentation and add tests for this rule if the behavior seems fine.

Docs and tests for `computed-property-spacing` have already been updated (this rule is now deprecated anyway).

`quote-props` and `key-spacing` are now deprecated, so no changes there.
